src/struct_utils.cc

```diff
@@ -45,10 +45,23 @@ StructUtils::FindResult StructUtils::GetUInt64(const std::string& name,
   if (found->kind_case() != google::protobuf::Value::kNumberValue) {
     return WRONG_TYPE;
   }
-  if (found->number_value() < 0) {
-    return NOT_POSITIVE;
+  *double_value = found->number_value();
+  return OK;
+}
+
+StructUtils::FindResult StructUtils::GetUInt64(const std::string& name,
+                                               uint64_t* int_value) {
+  double double_value;
+  FindResult result = GetDouble(name, &double_value);
+  if (result != OK) {
+    return result;
+  }
+  if (double_value < 0 ||
+      double_value >=
+          static_cast<double>(std::numeric_limits<uint64_t>::max())) {
+    return OUT_OF_RANGE;
   }
-  *int_value = static_cast<uint64_t>(found->number_value());
+  *int_value = static_cast<uint64_t>(double_value);
```

Comment:
will `double_value` have decimals? Seems they will be truncated. We should probably validate against decimals.

Reply:
I think it is expected that decimals be discarded after cast.  I can add a test for it

Reply:
Also add it to the function comment.

Reply:
Even without decimals, I don't think 64-bit `double` has enough precision to represent every 64-bit `uint64_t` integer values. It is capable to represent every 32-bit integers, but probably not 64-bit. 
Maybe try adding a test case that has int value `2^52 + 1` which might show the precision loss.

Reply:
This is the limitation of double.  I will just add a comment on the GetDouble function.

Reply:
Hmm, just googled it, found this [RFC](https://www.rfc-editor.org/rfc/rfc7159#section-6).  The spec. say that the range is restricted to "double" implementation which is  in  the range [-(2**53)+1, (2**53)-1]

So we are good.

Reply:
Do we need to preserve the precision beyond 32-bit integers? `double` can perfectly preserve any values a 32-bit `int` represents, but it can't for `uint64_t`. 
If not, we're ok, but if we do, we probably need another way to detect this big integer issue. Can we only cast `found->number_value()` without assigning it to `uint64_t` return value to detect big integer?

Reply:
GetUInt64 is used for "exp", "iat", dated related integers in this lib, I guess 32 bit is big enough.  But this function could be used for callers of this lib to extract any payload integer field,  we don't want to impose any restriction on these.